### PR TITLE
Add `Contract.interfaceId` field

### PR DESCRIPTION
**Pull request by tdroxler**

### CODE REVIEW

File: app/src/main/scala/org/alephium/explorer/persistence/model/ContractEntity.scala [hunk 1]
```diff
@@ -16,14 +16,17 @@
...
+
+import org.alephium.api.model.{ValAddress, ValByteVec}
 import org.alephium.protocol
 import org.alephium.protocol.model.{Address, BlockHash, TransactionId}
 import org.alephium.util.TimeStamp
 
 final case class ContractEntity(
     contract: Address,
     parent: Option[Address],
+    interfaceId: Option[ByteString],
```
Comment (polarker) [inline app/src/main/scala/org/alephium/explorer/persistence/model/ContractEntity.scala:29]: Should we name it explicitly that the id is just a guess and the id is for standard interface? Something like `stdInterfaceIdGuessed`.

File: app/src/main/scala/org/alephium/explorer/persistence/model/ContractEntity.scala [hunk 2]
```diff
@@ -78,18 +86,21 @@ object ContractEntity {
...
+  ): Option[(Address, Option[Address], Option[ByteString])] = {
+    if (event.fields.sizeIs == 3) {
+      (event.fields(0), event.fields(1), event.fields(2)) match {
+        case (ValAddress(contract), ValByteVec(ByteString.empty), ValByteVec(ByteString.empty)) =>
+          Some((contract, None, None))
+        case (ValAddress(contract), ValAddress(parent), ValByteVec(ByteString.empty)) =>
+          Some((contract, Some(parent), None))
+        case (ValAddress(contract), ValAddress(parent), ValByteVec(interfaceId))
+            if (interfaceId == createContractInterfaceIdPrefix) =>
+          Some((contract, Some(parent), Some(interfaceId)))
```
[the remark this answers is not in the document]
Comment (tdroxler) [reply app/src/main/scala/org/alephium/explorer/persistence/model/ContractEntity.scala:100]: empty cases are already handle aboved https://github.com/alephium/explorer-backend/blob/6aaf33215df90587ca2973664fde151b62480489/app/src/main/scala/org/alephium/explorer/persistence/model/ContractEntity.scala#L94-L97

so I guess I could just go with:
```scala
        case (ValAddress(contract), ValByteVec(ByteString.empty), ValByteVec(ByteString.empty)) =>
          Some((contract, None, None))
        case (ValAddress(contract), ValAddress(parent), ValByteVec(ByteString.empty)) =>
          Some((contract, Some(parent), None))
        case (ValAddress(contract), ValAddress(parent), ValByteVec(interfaceId))
          Some((contract, Some(parent), Some(interfaceId)))
        case (ValAddress(contract), ValByteVec(ByteString.empty), ValByteVec(interfaceId))
          Some((contract, None, Some(interfaceId)))
        case _ =>
          None

```
The 4th cases we spoke this morning

[the remark this answers is not in the document]
Comment (polarker) [reply app/src/main/scala/org/alephium/explorer/persistence/model/ContractEntity.scala:100]: Yes, just use the `id` directly

**Comment by tdroxler** [comment]
here is the `contracts` table after the migration:
```
testnet=# \d contracts
                           Table "public.contracts"
          Column          |       Type        | Collation | Nullable | Default
--------------------------+-------------------+-----------+----------+---------
 contract                 | character varying |           | not null |
 parent                   | character varying |           |          |
 creation_block_hash      | bytea             |           | not null |
 creation_tx_hash         | bytea             |           | not null |
 creation_timestamp       | bigint            |           | not null |
 creation_event_order     | integer           |           | not null |
 destruction_block_hash   | bytea             |           |          |
 destruction_tx_hash      | bytea             |           |          |
 destruction_timestamp    | bigint            |           |          |
 destruction_event_order  | integer           |           |          |
 std_interface_id_guessed | bytea             |           |          |
Indexes:
    "contracts_pk" PRIMARY KEY, btree (contract, creation_block_hash)
    "contracts_contract_idx" btree (contract)
    "contracts_parent_idx" btree (parent)
    "contracts_std_interface_id_guessed_idx" btree (std_interface_id_guessed)
```

**Review by Lbqds — APPROVED** [review]
LGTM 👍

**Pull request merged**